src/terra-dev-site/contributing/Conventions.d.contributing.md

```diff
@@ -0,0 +1,254 @@
+# Conventions
+
+## CSS Conventions
+
+* Use SCSS for styling components.
+* Follow [https://sass-guidelin.es/](https://sass-guidelin.es/) for component styles.
+* Avoid using the [CSS Modules `compose` feature](https://github.com/css-modules/css-modules#composition). This can start to blend cross-component concerns and we want each component to work independently.
+* Use CSS custom properties for theming. e.g. `var(--terra-button-color, #333);`
+  * If a theming property doesn't require a fallback, you can omit it from the CSS custom property. This will allow the browser to fallback to the inherited style for the property. For example, this code will make the button font-family themable, but use the currently inherited font-family value when the variable is not defined.
+  `var(--terra-button-font-family);`
+
+## JavaScript Conventions
+
+* Use JSX for React components.
+* Follow [Airbnb javascript style guide](https://github.com/airbnb/javascript).
+
+Below includes additional conventions we've adopted outside of the Airbnb javascript style guide.
+
+### Bind function to a component in constructor
+
+```jsx
+// Good
+constructor(props) {
+  super(props);
+  this.handleChange = this.handleChange.bind(this);
+}
+
+// Bad
+onChange={e => this.handleChange(e)}
+
+// Bad
+onChange={this.handleChange.bind(this)}
+```
+
+### Prefix event handler with `handle`
+
+```jsx
+// Good
+handleOnClick(e) {
+  const { onClick } = this.prop
+  onClick && onClick(e)
+}
+onClick={this.handleOnClick}
+
+// Bad
+onClick={onClick}
+```
+
+### Use functional component, if component doesn't require state or lifecycle hooks
+
+```jsx
+// Good
+const Component = (props) => {
+  ...
+  return <FinalReactComponent />;
+};
+
+// Bad
+class Component extends React.Component {
+  render() {
+    return <FinalReactComponent />;
+  }
+}
+```
+
+### React PropType Conventions
+
+#### Define propTypes and defaultProps before component implementation
+
+```jsx
+const propTypes = {
+ isDisabled: PropTypes.bool,
+ onClick: PropTypes.func,
+ text: PropTypes.string,
+ variant: PropTypes.oneOf(['link', 'outline']),
+};
+
+const defaultProps = {
+ isDisabled: false,
```

Comment:
Should you add variant, onClick and text defaults in this example? I think it would be valuable to make one of these required as we don't want defaults for required props

Reply:
Updated example: https://github.com/cerner/terra-ui/pull/108/commits/0120d495f4356a69257518f07acafcb9be89213d